src/infrastructure/middleware/authorization/whiteList/index.ts

```diff
@@ -4,17 +4,37 @@ import { BooleanUtil } from "../../../../domain/shared/utils/BooleanUtil";
 import { TypeParser } from "../../../../domain/shared/utils/TypeParser";
 import { Middleware } from "../../types";
 import config from "../../../config";
+import { StringCompare } from "./string.compare";
+import { RegExpCompare } from "./regexp.compare";
+import { PathEvaluation } from "./pathEvaluation";
 
-const ROUTE_WHITE_LIST = [
+const ROUTE_WHITE_LIST: Array<string | RegExp> = [
   `${config.Server.Root}/ping`,
   `${config.Server.Root}/v1/auth/login`,
   `${config.Server.Root}/v1/users/sign-up`,
+  "/docs",
 ];
 
 class RouteWhiteListMiddleware {

```

Comment:
In this context you have created a lot fo files and some of them are logic, so in that way I think they apply to be a domain utils because can be re-used in other context, but analyzing your implementation I think is a little bit complicated, so I design a new proposal for you and maybe you can perform it: 

```ts
const ROUTE_WHITE_LIST: { rule: "starts-with" | "equal"; value: string }[] = [
  { rule: "equal", value: `${config.Server.Root}/ping` },
  { rule: "equal", value: `${config.Server.Root}/v1/auth/login` },
  { rule: "equal", value: `${config.Server.Root}/v1/users/sign-up` },
  { rule: "starts-with", value: `${config.Server.Root}/docs` },
];

class RouteWhiteListMiddleware {
  handle: Middleware = (req: Request, _res: Response, next: NextFunction): void => {
    console.log("RouteWhiteListMiddleware", req.path);
    const existsUnauthorizedPath = ROUTE_WHITE_LIST.some((route) =>
      route.rule === "equal"
        ? BooleanUtil.areEqual(route.value, req.path)
        : req.path.startsWith(route.value),
    );

    TypeParser.cast<IRequest>(req).isWhiteList = existsUnauthorizedPath;

    return next();
  };
}
```

Reply:
Liked, removed my logic and committed yours